src/core/pointcloud/qgspointcloudeditingindex.cpp

```diff
@@ -138,5 +196,19 @@ bool QgsPointCloudEditingIndex::updateNodeData( const QHash<QgsPointCloudNodeId,
     mEditedNodeData[it.key()] = it.value();
   }
 
+  const QList<QgsPointCloudNodeId> modifiedNodesIds = data.keys();
+  QSet<QgsPointCloudNodeId> modifiedNodesSet( modifiedNodesIds.constBegin(), modifiedNodesIds.constEnd() );
+
+  // get rid of cached keys that got modified
+  {
+    QMutexLocker locker( &sBlockCacheMutex );
+    const QList<QgsPointCloudCacheKey> cacheKeys = sBlockCache.keys();
+    for ( QgsPointCloudCacheKey cacheKey : cacheKeys )
+    {
+      if ( cacheKey.uri() == mUri && modifiedNodesSet.contains( cacheKey.node() ) )
+        sBlockCache.remove( cacheKey );
+    }
+  }
```

Comment:
`QHash::keys()` creates a copy of the keys.
Could we use `data.contains( cacheKey.node() )` instead of going though QList and QSet?

Then for the second case of `sBlockCache.keys()` we could use a hash iterator and `QHash::erase()`?